### PR TITLE
feat(BigcommerceProductAgent): enable product creation as Variants

### PR DESCRIPTION
**Pull request by jspizziri**
@mrogers-5s , I think I've done most of the scaffolding work, however I haven't run or tested it at all. I'll try to pick it up next week, but I might need to hand it off to you. I'll let you know if I do.

### CODE REVIEW

**Comment by jspizziri** [comment]
@mrogers-5s I've tested this and it seems to be working for the basic scenario's I could think of. It's not pretty, but it looks like it get's the job done. Could probably use some refactoring/abstraction, but it also might be good to get it installed to open up more testing for you guys.

File: lib/mapper/product_mapper.rb [hunk 1]
```diff
@@ -68,6 +76,36 @@ def self.get_is_default(product)
...
+
+                digitals = product['model'].select {|m| m['isDigital'] != true}
+
+                if digitals.length > 0
+                    clone = Marshal.load(Marshal.dump(product))
+                    clone['model'] = digitals
+                    result[:digital] = clone
+                end
+
+                physicals = product['model'].select {|m| m['isDigital'] == true}
```
Comment (mrogers-5s) [inline lib/mapper/product_mapper.rb:98]: @jspizziri Looks like these are flopped

Comment (jspizziri) [reply lib/mapper/product_mapper.rb]: @mrogers-5s doh!

Fixed.

**Comment by Jhalter5Stones** [comment]
@jspizziri Do you think we could get this released to staging so the we can start getting more testing for a larger variety of products?

**Comment by jspizziri** [comment]
@Jhalter5Stones released as `v1.3.0`

**Pull request merged**